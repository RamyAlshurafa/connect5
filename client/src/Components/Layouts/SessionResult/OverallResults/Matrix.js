import React from "react";
import PropTypes from "prop-types";
import { MatrixAnswer } from "../StyledComponents";

const Matrix = ({ answers }) => (
  <div>
    <ul>
      {
        answers && answers[0].answer.map(item => (
          <li key={Math.random()}>
            <MatrixAnswer>
              {item}
            </MatrixAnswer>
          </li>
        ))
      }
    </ul>
  </div>
);

export default Matrix;

Matrix.propTypes = {
  answers: PropTypes.arrayOf(PropTypes.shape({
    answer: PropTypes.arrayOf(PropTypes.string),
  })),
};
Matrix.defaultProps = {
  answers: [{}],
};
