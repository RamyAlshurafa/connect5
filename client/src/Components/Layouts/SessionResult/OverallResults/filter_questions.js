const filterquestions = (questions) => {
  console.log(questions.slice(5 - questions.length));

  return questions.slice(5 - questions.length);
};
export default filterquestions;
