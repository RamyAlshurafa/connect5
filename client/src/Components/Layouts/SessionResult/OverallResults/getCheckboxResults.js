

const getCheckboxResults = (answers, options) => {

  // create total list of options to include potential other answers

  const totalOptions = options.map( option => option );
  const newAnswers = answers;
  const output = [];

  answers.map(answer => {
    if (!options.includes(answer)) {
      totalOptions.push(answer)
    }
    })

    totalOptions.map((option) => {
      const cummulative = answers.reduce((cumm, answer) => {
        if (answer === option) {
          return cumm + 1;
        }
        return cumm;
      }, 0);
      return output.push({ value: option, cummulative });
    });

    return { newOptions: output, newAnswers };

}

export default getCheckboxResults;