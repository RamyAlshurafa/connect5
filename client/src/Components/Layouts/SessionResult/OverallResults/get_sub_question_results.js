const getSubQuestionResult = (columns, rows, answers) => {
  const output = {};
  rows.map((row) => {
    const columnsWithCumm = {};
    columns.map((item) => {
      columnsWithCumm[item] = 0;
    });
    output[row] = columnsWithCumm;
  });


  answers.map((inputanswer) => {
    const inputQuestions = Object.keys(inputanswer.answer);

    inputQuestions.map((inputQuestion) => {
      const inputColumn = inputanswer.answer[inputQuestion];

      output[inputQuestion][inputColumn] = output[inputQuestion][inputColumn] + 1;
    });
  });
  return output;
};

export default getSubQuestionResult;
