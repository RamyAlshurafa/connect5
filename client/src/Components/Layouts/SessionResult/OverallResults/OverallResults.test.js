import React from "react";
import renderer from "react-test-renderer";
import { StaticRouter } from "react-router-dom";

import OverallResults from "./index";

test("snapshot for 'OverallResults' layout", () => {
  const context = {};
  const match = {
    params: {
      sessionId: "sessionId",
      sessionType: "sessionType",
    },
  };

  const component = renderer.create(
    <StaticRouter context={context}>
      <OverallResults match={match} />
    </StaticRouter>,
  );
  expect(component).toMatchSnapshot();
});
