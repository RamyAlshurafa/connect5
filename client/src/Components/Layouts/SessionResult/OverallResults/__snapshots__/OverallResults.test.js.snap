// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot for 'OverallResults' layout 1`] = `
<div>
  <div
    className="sc-ifAKCX gVwUCl"
  >
    <p
      className="sc-EHOje hKuwFc"
    >
      <span
        className="sc-bZQynM kYemYx"
      >
        Number of Replies:
      </span>
       
      (out of
       
       
      )
    </p>
    <p
      className="sc-EHOje hKuwFc"
    >
      <span
        className="sc-bZQynM kYemYx"
      >
        Completion Rate:
      </span>
      NaN
      %
    </p>
  </div>
  <h3
    className="sc-gzVnrw gdyVTx"
  >
    Responses
  </h3>
</div>
`;
